sdk/src/main/scala/com/horizen/forge/ForgingControl.scala

```diff
@@ -0,0 +1,90 @@
+package com.horizen.forge
+
+import java.time.Instant
+import java.util.concurrent.Executors
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.util.Timeout
+import com.horizen._
+import com.horizen.block.SidechainBlock
+import com.horizen.consensus.TimeToEpochSlotConverter
+import com.horizen.forge.Forger.SendMessages.{ForgeResult, ForgeSuccess}
+import com.horizen.params.NetworkParams
+import scorex.core.NodeViewHolder.ReceivableMessages.LocallyGeneratedModifier
+import scorex.util.ScorexLogging
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Success
+
+class ForgingControl(settings: SidechainSettings, forgerRef: ActorRef, viewHolderRef: ActorRef, val params: NetworkParams) extends Actor with ScorexLogging with TimeToEpochSlotConverter {
```

Comment:
1. Additional `ForgingControl` Actor looks like an overhead. It's something that has a behavior of both `Forger` and `BlockActor`. Moreover, now both `ForgingControl` and `BlockActor` can submit a Block, but `Forger` can't.
2. Automatic forging can be implemented similar to Scorex Examples: they use only Actor messages system. So, `@volatile private var forgingIsActive` access can be controlled just by Actor messages processing flow.